cirq/protocols/json_serialization.py

```diff
@@ -440,8 +428,9 @@ def _cirq_object_hook(d, resolvers: Sequence[JsonResolver]):
         raise ValueError("Could not resolve type '{}' "
                          "during deserialization".format(d['cirq_type']))
 
-    if hasattr(cls, '_from_json_dict_'):
-        return cls._from_json_dict_(**d)
+    from_json_dict = getattr(cls, '_from_json_dict_', None)
+    if from_json_dict is not None:
```

Comment:
why the change?

Reply:
The old code doesn't type check because mypy doesn't understand that `cls._from_json_dict_` is valid after the `hasattr` check.